sdk/eventgrid/eventgrid/review/eventgrid.api.md

```diff
@@ -10,15 +10,47 @@ import { OperationOptions } from '@azure/core-http';
 import { PipelineOptions } from '@azure/core-http';
 import { RestResponse } from '@azure/core-http';
 
+// @public
+export type AppAction = "Restarted" | "Stopped" | "ChangedAppSettings" | "Started" | "Completed" | "Failed";
+
+// @public
+export interface AppConfigurationKeyValueDeletedEventData {
+    etag?: string;
+    key?: string;
```

Comment:
is it really possible that key could be undefined? I would think at least one property would be required...

Reply:
I opened https://github.com/Azure/azure-sdk-for-js/issues/10992 to explain what is going on here and to see if we can fix it for preview2.  The long and short of it is that we generate these interfaces from specs that the individual service teams author and so while I think you are correct that we can be more strict here, I want to take some time and understand what guarantees the service team is actually providing so we can reflect it in our typings.  Does that sound reasonable?

Reply:
I think that's reasonable as long as we can get to a good place before GA (as making an optional property required is technically breaking)